pkg/models/kubeconfig/kubeconfig.go

```diff
@@ -21,295 +21,228 @@ package kubeconfig
 import (
 	"bytes"
 	"crypto/rand"
-	"crypto/rsa"
 	"crypto/x509"
-	"crypto/x509/pkix"
-	"encoding/base64"
 	"encoding/pem"
 	"fmt"
-	"gopkg.in/yaml.v2"
-	"io/ioutil"
+	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	certutil "k8s.io/client-go/util/cert"
 	"k8s.io/klog"
+	"kubesphere.io/kubesphere/pkg/constants"
+	pkiutil "kubesphere.io/kubesphere/pkg/models/kubeconfig/internal"
 	"kubesphere.io/kubesphere/pkg/simple/client"
-	"math/big"
-	rd "math/rand"
 	"time"
-
-	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
-	"k8s.io/apimachinery/pkg/api/errors"
-
-	"k8s.io/api/core/v1"
-
-	"kubesphere.io/kubesphere/pkg/constants"
 )
 
 const (
-	caPath           = "/etc/kubernetes/pki/ca.crt"
-	keyPath          = "/etc/kubernetes/pki/ca.key"
-	clusterName      = "kubernetes"
-	kubectlConfigKey = "config"
-	defaultNamespace = "default"
+	kubeconfigNameFormat = "kubeconfig-%s"
+	defaultClusterName   = "cluster.local"
```

Comment:
does this cluster name mean something? why it have to be `cluster.local`, same with the default domain name, it's misleading